src/html-module-transform.ts

```diff
@@ -51,7 +51,7 @@ export const htmlModuleToJsModuleMap =
 
       if (property.type === 'Identifier') {
```

Comment:
I think the problem here is that this check is not specific enough. The condition is checking for usages of `import.meta.*` as an expression, but isn't particular about `import.meta.script` (the original use case) vs `import.meta.document`.

We should probably change the condition to something like:

```typescript
if (property.type === 'Identifier') {
  switch (property.name) {
    case 'script':
      // Do the current transform for script
      break;
    case 'document':
      // Do the new proposed transform for document
      break;
  }
}
```